### DISCUSSION TITLE
Calling a procedure outside of React

### DISCUSSION THREAD

**Discussion by ThePaulMcBride** [trpc/trpc#3736] [Q&A]
I have a Next.js app setup the way [this guide](https://trpc.io/docs/nextjs) recommends.

The problem I'm facing is that I can't seem to figure out how to call a procedure outside of React as all I have access to are the hooks exposed by `createTRPCNext`. Is there a way to do this?

**Comment by arosemena** [comment]
If you inspect the network tab and do an action with a query you'll see the format that trpc uses, you could imitate that and do a regular fetch request with the same pattern from outside your codebase.

Here's some more info on the specification: https://trpc.io/docs/rpc

**Comment by ThePaulMcBride** [comment]
I've been able to o something similar and get this working, but is there a way to do it so that I still get all of the type safety offered by TRPC?

**Comment by arosemena** [comment]
Have you looked into the fetch adapter?

https://trpc.io/docs/fetch

Where exactly are you trying to use it from?

Checks the adapters from this page maybe there's one for your use case:
https://trpc.io/docs/awesome-trpc#library-adapters

**Comment by tgdn** [comment]
You should use [`createTRPCQueryUtils`](https://trpc.io/docs/client/react/createTRPCQueryUtils)
